aten/src/ATen/templates/TypeDerived.cpp

```diff
@@ -91,11 +91,7 @@ Storage ${Type}::storageWithAllocator(int64_t size, Allocator* allocator) const
     return Storage(ScalarType::${ScalarName}, size, allocator);
 }
 Tensor ${Type}::unsafeTensorFromTH(void * th_pointer, bool retain) const {
-  TensorImpl* pimpl = (TensorImpl*)(th_pointer);
-  if (retain) {
-    pimpl->retain();
-  }
-  return Tensor(pimpl, false);
+  return Tensor(static_cast<TensorImpl*>(th_pointer), retain);
```

Comment:
this will retain an UndefinedTensor where it didn't before -- do we want that?

Reply:
It's OK, because the `intrusive_ptr` codepath no-ops in this case. (It's a little hard to see, but we incref by reclaiming a pointer, and then making a copy of the `intrusive_ptr`. Reclaiming UndefinedTensor* used to be illegal but I relaxed the assert in `intrusive_ptr`, and copies of `intrusive_ptr`, when an `UndefinedTensor` is pointed at, are just no-ops, as you would expect.)